### ISSUE TITLE
WEBUI renew

### ISSUE THREAD

**Issue by ffdown** [AlekPet/Fooocus_Extensions_AlekPet#2]
новых мыслишек не пришло как перепришпандорить? А то фукус обновляться пока любит, и webui периодически перезаписывается..

**Comment by igor469** [comment]
Пытался сам вставлять куски в новые редакции webui.py по подобию авторского
До версии 2.1.83 такое работало, а теперь тоже интересует как решить эту проблемму.
А автор здесь бывает?

**Comment by ffdown** [comment]
> Пытался сам вставлять куски в новые редакции webui.py по подобию авторского До версии 2.1.83 такое работало, а теперь тоже интересует как решить эту проблемму. А автор здесь бывает?

сомнительно... это ж тот же автор, что и в целом controlnet замутил

**Comment by AlekPet** [comment]
Автор тут. Обновился фокус?

**Comment by igor469** [comment]
Да, обновился Foocus до  2.1.841 и новый prompt_translate отлично работает (!) Сенкс

**Comment by igor469** [comment]
Версия 2.1.855 вновь сильно поменяла webui.py и prompt_translate исчез
Сможете поправить или пытаться самому аккуратно переставлять куски из предыдущей редакции ?

**Comment by igor469** [comment]
Версия 2.1.856 
Переставил куски "Prompt translate AlekPet"
Кнопка "Translate" работает хорошо, а вот "Auto translate "Prompt and Negative prompt" before Generate" - увы,
не получилось

**Comment by AlekPet** [comment]
Создал патчер для установки расширения, чтобы он сам обновлял webui.py, когда обновился Fooocus.
[patcher-webui.py](https://github.com/AlekPet/Fooocus_Extensions_AlekPet/tree/main?tab=readme-ov-file#patcher-webui)

**Comment by igor469** [comment]
Только что провер работу патчера на версии 2.1.862 - работает как часы
Были похожие чудеса, когда менял туда-сюда entry_with_update.py и webui.py для автогенерации 1 своей картинке на каждый стиль Fooocus-a и не смог правильно вернуться все назад.
Тогда просто скачал свежие entry_with_update.py, webui.py и все снова заработало

**Comment by kaarst90** [comment]
thank you my fooocus not running cause your code changed the default even after i delete your "google" translator" <3

**Comment by AlekPet** [comment]
> thank you my fooocus not running cause your code changed the default even after i delete your "google" translator" <3

Since fooocus does not have plugin support, it is possible that the code will change and the focus may not work, you can always download webui.py from the original repository and replace it.  If there was support for dancing with a tambourine, there would be no such thing, but anything can happen.

**Comment by AlekPet** [comment]
Update patcher for new version: Fooocus version: 2.2.1

**Comment by Greshnic07** [comment]
2.3.1. версия фокуса, не работает с данным расширением(( Ошибка: 
AttributeError: module 'modules.html' has no attribute 'css'

**Comment by AlekPet** [comment]
Привет мне уже писали на почту про похожую ошибку. Проверь, чтобы мой патчер в bat или sh файле был первой строкой, а не второй, а также сбрось webui.py на оригинальный файл, либо через гит (git restore path_to_webui/webui.py) или просто скачай с гитхаба и замени. Вот скрин с версией 2.3.1
![20240430_184040.jpg](https://github.com/AlekPet/Fooocus_Extensions_AlekPet/assets/25489996/0122a6c9-69dc-46b9-a81d-bf2dcd11254b)
